### PR TITLE
Improve the dummy country to enable more tests on variable API

### PR DESCRIPTION
**Pull request by fpagnoux**
Connected to openfisca/openfisca-web-api#125

### CODE REVIEW

File: openfisca_dummy_country/model/model.py [hunk 1]
```diff
@@ -165,15 +175,60 @@ def function(individu, period):
...
-class contribution_sociale(Variable):
+class contribution_sociale(DatedVariable):
     column = FloatCol
     entity = Individu
     label = u"Contribution payée sur le salaire"
     definition_period = YEAR
 
-    def function(individu, period, legislation):
+    @dated_function(start = datetime.date(1880, 1, 1))
+    def function_1880(individu, period, legislation):
```
Comment (MattiSG) [inline openfisca_dummy_country/model/model.py:185]: > Why do we go back in time so much? This is fun, but might be misleading for new users not understanding the actual scope of OpenFisca.

File: openfisca_dummy_country/model/model.py [hunk 2]
```diff
@@ -165,15 +175,60 @@ def function(individu, period):
...
+    stop_date = date(1789, 12, 31)
+
+    def function(individu, period, legislation):
+        return individu('age') >= 18 * 400
+
+
+class api(DatedVariable):
+    column = FloatCol
+    entity = Famille
+    label = u"Allocationd pour Parent Isolé"
```
Comment (MattiSG) [inline openfisca_dummy_country/model/model.py]: Typo.

File: openfisca_dummy_country/model/model.py [hunk 3]
```diff
@@ -134,7 +131,20 @@ def function_2011_2012(individu, period):
...
+        return (salaire_imposable < 500) * 300.0
+
+
+class rmi(DatedVariable):
+    column = FloatCol
+    entity = Individu
+    label = u"RMI (remplacé par le RSA en 2010)"
+    definition_period = MONTH
+
+    @dated_function(datetime.date(2000, 1, 1), datetime.date(2009, 12, 31))
```
Comment (MattiSG) [inline openfisca_dummy_country/model/model.py]: Inconsistency: later, this is used as `start = `, and here the parameters are only indexed.

**Review by MattiSG — CHANGES_REQUESTED** [review]

File: openfisca_dummy_country/model/model.py (hunk 3 above)
Comment (fpagnoux) [reply openfisca_dummy_country/model/model.py]: 👍

File: openfisca_dummy_country/model/model.py (hunk 1 above)
Comment (fpagnoux) [reply openfisca_dummy_country/model/model.py:185]: There was a bug with pre-1900 dates. Should this not be tested ?

Comment (MattiSG) [reply openfisca_dummy_country/model/model.py:185]: Ok, but perhaps not with so many, so as not to be misleading.

**Review by MattiSG — APPROVED** [review]

**Pull request merged**